optimizely/notification_center.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2017, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+
+from .helpers import enums
+
+
+class NotificationCenter(object):
+  """ Class encapsulating Broadcast Notifications. The enums.NotifcationTypes includes predefined notifications."""
+  def __init__(self, logger):
+    self.notification_id = 1
+    self.notifications = {}
+    for attr, value in enums.NotificationTypes.__dict__.items():
+      self.notifications[value] = []
+    self.logger = logger
+
+  def add_notification(self, notification_type, notification_callback):
+    """ Add a notification callback to the notification center.
+
+    Args:
+      notification_type: .helpers.enums.NotificationTypes
+      notification_callback: closure of function to call when event is triggered.
+    Returns:
+      notification id used to remove the notification
+    """
+    if notification_type not in self.notifications:
+      self.notifications[notification_type] = [(self.notification_id, notification_callback)]
+    else:
+      for callback in self.notifications[notification_type]:
+        if callback[1] == notification_callback:
```

Comment:
A cleaner way of accessing the callback would be to unpack the tuple in the `for` declaration like this:

`for notification_id, callback in self.notifications[notification_type]:`